{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'d:\\\\Thesis_results'"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "# Change directory\n",
    "os.chdir('d:\\\\Thesis_results')\n",
    "\n",
    "# Now the current working directory is 'd:\\\\Thesis_results'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "# Define the directory path\n",
    "directory_path = 'd:\\\\Thesis_results\\\\snps.tabs\\\\'\n",
    "\n",
    "# Create a list of file paths\n",
    "files = [os.path.join(directory_path, f) for f in os.listdir(directory_path) if f.endswith(\".tab\")]\n",
    "\n",
    "# Read and concatenate the snp.tab files\n",
    "df_list = [pd.read_csv(f, sep='\\t') for f in files]\n",
    "df = pd.concat(df_list)\n",
    "\n",
    "# Reset index if needed\n",
    "df.reset_index(drop=True, inplace=True)\n",
    "\n",
    "# Remove duplicates\n",
    "df = df.drop_duplicates()\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>EVIDENCE</th>\n",
       "      <th>FTYPE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>NT_POS</th>\n",
       "      <th>AA_POS</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>LOCUS_TAG</th>\n",
       "      <th>GENE</th>\n",
       "      <th>PRODUCT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1849</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>A:139 C:0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1977</td>\n",
       "      <td>snp</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G:80 A:0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3446</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T:145 C:0</td>\n",
       "      <td>CDS</td>\n",
       "      <td>+</td>\n",
       "      <td>167/1158</td>\n",
       "      <td>56/385</td>\n",
       "      <td>missense_variant c.167C&gt;T p.Ala56Val</td>\n",
       "      <td>Rv0003</td>\n",
       "      <td>recF</td>\n",
       "      <td>DNA replication/repair protein RecF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>4013</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C:122 T:0</td>\n",
       "      <td>CDS</td>\n",
       "      <td>+</td>\n",
       "      <td>734/1158</td>\n",
       "      <td>245/385</td>\n",
       "      <td>missense_variant c.734T&gt;C p.Ile245Thr</td>\n",
       "      <td>Rv0003</td>\n",
       "      <td>recF</td>\n",
       "      <td>DNA replication/repair protein RecF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>7362</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>C:141 G:0</td>\n",
       "      <td>CDS</td>\n",
       "      <td>+</td>\n",
       "      <td>61/2517</td>\n",
       "      <td>21/838</td>\n",
       "      <td>missense_variant c.61G&gt;C p.Glu21Gln</td>\n",
       "      <td>Rv0006</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>DNA gyrase subunit A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM   POS TYPE REF ALT   EVIDENCE FTYPE STRAND    NT_POS   AA_POS  \\\n",
       "0  NC_000962  1849  snp   C   A  A:139 C:0   NaN    NaN       NaN      NaN   \n",
       "1  NC_000962  1977  snp   A   G   G:80 A:0   NaN    NaN       NaN      NaN   \n",
       "2  NC_000962  3446  snp   C   T  T:145 C:0   CDS      +  167/1158   56/385   \n",
       "3  NC_000962  4013  snp   T   C  C:122 T:0   CDS      +  734/1158  245/385   \n",
       "4  NC_000962  7362  snp   G   C  C:141 G:0   CDS      +   61/2517   21/838   \n",
       "\n",
       "                                  EFFECT LOCUS_TAG  GENE  \\\n",
       "0                                    NaN       NaN   NaN   \n",
       "1                                    NaN       NaN   NaN   \n",
       "2   missense_variant c.167C>T p.Ala56Val    Rv0003  recF   \n",
       "3  missense_variant c.734T>C p.Ile245Thr    Rv0003  recF   \n",
       "4    missense_variant c.61G>C p.Glu21Gln    Rv0006  gyrA   \n",
       "\n",
       "                               PRODUCT  \n",
       "0                                  NaN  \n",
       "1                                  NaN  \n",
       "2  DNA replication/repair protein RecF  \n",
       "3  DNA replication/repair protein RecF  \n",
       "4                 DNA gyrase subunit A  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'d:\\\\Thesis_results'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd\n",
    "#low = pd.read_csv(\"LOW_CD4_merged_variants.txt\", sep='\\t')\n",
    "low = pd.read_csv(\"HIGH_CD4_merged_variants.txt\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "low.shape#_4743 variants among low CD4 cell individuals\n",
    "#1326 variants among individuals with high and low CD4 cell count\n",
    "low['TYPE'].value_counts().to_csv(\"Low_CD4_results/SNP_TYPE.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>Effect</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Product</th>\n",
       "      <th>Gene_mutation</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7535067</th>\n",
       "      <th>SRR7535083</th>\n",
       "      <th>SRR7535084</th>\n",
       "      <th>SRR7535087</th>\n",
       "      <th>SRR7535088</th>\n",
       "      <th>SRR7535089</th>\n",
       "      <th>SRR7535090</th>\n",
       "      <th>SRR7535092</th>\n",
       "      <th>SRR7535093</th>\n",
       "      <th>Frequency_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>412</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0001</td>\n",
       "      <td>dnaA</td>\n",
       "      <td>chromosomal replication initiator protein DnaA</td>\n",
       "      <td>c.412G&gt;A</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1778</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv0001-Rv0002</td>\n",
       "      <td>dnaA-dnaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>n.1778C&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1908</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv0001-Rv0002</td>\n",
       "      <td>dnaA-dnaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>n.1908A&gt;C</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2883</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0002</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>DNA polymerase III subunit beta</td>\n",
       "      <td>c.832G&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>4572</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0004</td>\n",
       "      <td>Rv0004</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>c.139C&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>5306</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv0005</td>\n",
       "      <td>gyrB</td>\n",
       "      <td>DNA gyrase subunit B</td>\n",
       "      <td>c.67C&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>5506</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv0005</td>\n",
       "      <td>gyrB</td>\n",
       "      <td>DNA gyrase subunit B</td>\n",
       "      <td>c.267C&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>5524</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv0005</td>\n",
       "      <td>gyrB</td>\n",
       "      <td>DNA gyrase subunit B</td>\n",
       "      <td>c.285C&gt;A</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>7582</td>\n",
       "      <td>ACAG</td>\n",
       "      <td>CCAC</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0006</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>DNA gyrase subunit A</td>\n",
       "      <td>c.281_284delACAGinsCCAC</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>9596</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv0006</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>DNA gyrase subunit A</td>\n",
       "      <td>c.2295G&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM   POS   REF   ALT        Variant_type    Effect       Locus_id  \\\n",
       "0  NC_000962   412     G     A    missense_variant  MODERATE         Rv0001   \n",
       "1  NC_000962  1778     C     T   intergenic_region  MODIFIER  Rv0001-Rv0002   \n",
       "2  NC_000962  1908     A     C   intergenic_region  MODIFIER  Rv0001-Rv0002   \n",
       "3  NC_000962  2883     G     T    missense_variant  MODERATE         Rv0002   \n",
       "4  NC_000962  4572     C     T    missense_variant  MODERATE         Rv0004   \n",
       "5  NC_000962  5306     C     T  synonymous_variant       LOW         Rv0005   \n",
       "6  NC_000962  5506     C     T  synonymous_variant       LOW         Rv0005   \n",
       "7  NC_000962  5524     C     A  synonymous_variant       LOW         Rv0005   \n",
       "8  NC_000962  7582  ACAG  CCAC    missense_variant  MODERATE         Rv0006   \n",
       "9  NC_000962  9596     G     T  synonymous_variant       LOW         Rv0006   \n",
       "\n",
       "     Gene_id                                         Product  \\\n",
       "0       dnaA  chromosomal replication initiator protein DnaA   \n",
       "1  dnaA-dnaN                                             NaN   \n",
       "2  dnaA-dnaN                                             NaN   \n",
       "3       dnaN                 DNA polymerase III subunit beta   \n",
       "4     Rv0004                            hypothetical protein   \n",
       "5       gyrB                            DNA gyrase subunit B   \n",
       "6       gyrB                            DNA gyrase subunit B   \n",
       "7       gyrB                            DNA gyrase subunit B   \n",
       "8       gyrA                            DNA gyrase subunit A   \n",
       "9       gyrA                            DNA gyrase subunit A   \n",
       "\n",
       "             Gene_mutation  ... SRR7535067 SRR7535083  SRR7535084  SRR7535087  \\\n",
       "0                 c.412G>A  ...          0          0           0           0   \n",
       "1                n.1778C>T  ...          0          0           0           0   \n",
       "2                n.1908A>C  ...          0          0           0           1   \n",
       "3                 c.832G>T  ...          0          0           0           1   \n",
       "4                 c.139C>T  ...          0          0           0           0   \n",
       "5                  c.67C>T  ...          0          0           1           0   \n",
       "6                 c.267C>T  ...          0          0           0           0   \n",
       "7                 c.285C>A  ...          0          0           0           0   \n",
       "8  c.281_284delACAGinsCCAC  ...          0          0           0           0   \n",
       "9                c.2295G>T  ...          0          0           0           1   \n",
       "\n",
       "   SRR7535088  SRR7535089  SRR7535090  SRR7535092  SRR7535093  Frequency_Count  \n",
       "0           0           0           0           0           0                1  \n",
       "1           0           0           0           0           1                1  \n",
       "2           0           0           0           0           0                1  \n",
       "3           0           0           0           0           0                1  \n",
       "4           0           0           0           0           0                1  \n",
       "5           0           0           0           0           0                1  \n",
       "6           0           0           0           0           0                1  \n",
       "7           0           0           0           0           0                1  \n",
       "8           0           0           0           1           0                1  \n",
       "9           0           0           0           0           0                1  \n",
       "\n",
       "[10 rows x 30 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "low.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Assuming df is your original DataFrame\n",
    "\n",
    "# Create a new DataFrame containing rows with NaN values\n",
    "intergenic = low[low.isna().any(axis=1)].copy()\n",
    "\n",
    "# Reset index if needed\n",
    "intergenic.reset_index(drop=True, inplace=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(164, 30)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intergenic.head()\n",
    "intergenic.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                      Variant_type  high_count\n",
      "0                                 missense_variant         669\n",
      "1                               synonymous_variant         395\n",
      "2                                intergenic_region         159\n",
      "3                               frameshift_variant          51\n",
      "4                                      stop_gained          16\n",
      "5                    conservative_inframe_deletion           7\n",
      "6                      disruptive_inframe_deletion           6\n",
      "7                   conservative_inframe_insertion           6\n",
      "8                    non_coding_transcript_variant           3\n",
      "9                  stop_lost&splice_region_variant           3\n",
      "10                    disruptive_inframe_insertion           2\n",
      "11     splice_region_variant&stop_retained_variant           2\n",
      "12                  frameshift_variant&stop_gained           2\n",
      "13             frameshift_variant&missense_variant           2\n",
      "14           frameshift_variant&synonymous_variant           1\n",
      "15  disruptive_inframe_deletion&synonymous_variant           1\n",
      "16                                      start_lost           1\n"
     ]
    }
   ],
   "source": [
    "# Count all unique values in the 'EFFECT' column\n",
    "effect_counts = low['Variant_type'].value_counts()\n",
    "\n",
    "# Create a DataFrame with 'EFFECT' and 'low_count'\n",
    "effect_df_1 = pd.DataFrame({'Variant_type': effect_counts.index, 'high_count': effect_counts.values})\n",
    "\n",
    "# Print the counts\n",
    "print(effect_df_1)\n",
    "\n",
    "# Save the DataFrame to a CSV file\n",
    "effect_df_1.to_csv('High_CD4_results/effect_count.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fill empty 'Gene' values with corresponding 'LocusID'\n",
    "low['Gene_id'].fillna(low['Locus_id'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_panel = pd.read_csv(\"resistance genes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(41, 6)"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_panel.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>WHO category</th>\n",
       "      <th>Drug or drug class</th>\n",
       "      <th>Resistance genes</th>\n",
       "      <th>Rv number</th>\n",
       "      <th>Gene function</th>\n",
       "      <th>Mechanism of drug resistance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>First-line agents</td>\n",
       "      <td>Rifamycins (for example, rifampicin)</td>\n",
       "      <td>rpoB</td>\n",
       "      <td>Rv0667</td>\n",
       "      <td>RNA polymerase</td>\n",
       "      <td>Target modification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoA</td>\n",
       "      <td>Rv0667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ponA1</td>\n",
       "      <td>Rv0050</td>\n",
       "      <td>Probable bifunctional penicillin-binding protein</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Isoniazid</td>\n",
       "      <td>katG</td>\n",
       "      <td>Rv1908c</td>\n",
       "      <td>Catalase-peroxidase enzyme</td>\n",
       "      <td>Decreased drug activation</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        WHO category                    Drug or drug class Resistance genes  \\\n",
       "0  First-line agents  Rifamycins (for example, rifampicin)             rpoB   \n",
       "1                NaN                                   NaN             rpoA   \n",
       "2                NaN                                   NaN             rpoC   \n",
       "3                NaN                                   NaN            ponA1   \n",
       "4                NaN                             Isoniazid             katG   \n",
       "\n",
       "  Rv number                                     Gene function  \\\n",
       "0    Rv0667                                    RNA polymerase   \n",
       "1    Rv0667                                               NaN   \n",
       "2       NaN                                               NaN   \n",
       "3    Rv0050  Probable bifunctional penicillin-binding protein   \n",
       "4   Rv1908c                        Catalase-peroxidase enzyme   \n",
       "\n",
       "  Mechanism of drug resistance  \n",
       "0          Target modification  \n",
       "1                          NaN  \n",
       "2                          NaN  \n",
       "3                      Unknown  \n",
       "4    Decreased drug activation  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_panel.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_frequency = low['Gene_id'].value_counts().reset_index()\n",
    "gene_frequency.columns = ['Resistance genes', 'Frequency']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1071, 2)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_frequency.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Resistance genes</th>\n",
       "      <th>Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>phoH2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Rv1084</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PPE8</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>eccC2</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bglS</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Resistance genes  Frequency\n",
       "0            phoH2          6\n",
       "1           Rv1084          5\n",
       "2             PPE8          5\n",
       "3            eccC2          5\n",
       "4             bglS          4"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_frequency.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "resistance_genes = gene_panel[gene_panel['Resistance genes'].isin(low['Gene_id'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(11, 6)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resistance_genes.shape # 11 genes associated with drug resistance in HIGH CD4 cell counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>WHO category</th>\n",
       "      <th>Drug or drug class</th>\n",
       "      <th>Resistance genes</th>\n",
       "      <th>Rv number</th>\n",
       "      <th>Gene function</th>\n",
       "      <th>Mechanism of drug resistance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ponA1</td>\n",
       "      <td>Rv0050</td>\n",
       "      <td>Probable bifunctional penicillin-binding protein</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Isoniazid</td>\n",
       "      <td>katG</td>\n",
       "      <td>Rv1908c</td>\n",
       "      <td>Catalase-peroxidase enzyme</td>\n",
       "      <td>Decreased drug activation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpsA</td>\n",
       "      <td>Rv1630</td>\n",
       "      <td>Ribosomal protein S1</td>\n",
       "      <td>Target modification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Group A</td>\n",
       "      <td>Levofloxacin</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>Rv0006</td>\n",
       "      <td>DNA gyrase A</td>\n",
       "      <td>Target modification</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   WHO category Drug or drug class Resistance genes Rv number  \\\n",
       "2           NaN                NaN             rpoC       NaN   \n",
       "3           NaN                NaN            ponA1    Rv0050   \n",
       "4           NaN          Isoniazid             katG   Rv1908c   \n",
       "8           NaN                NaN             rpsA    Rv1630   \n",
       "11      Group A       Levofloxacin             gyrA    Rv0006   \n",
       "\n",
       "                                       Gene function  \\\n",
       "2                                                NaN   \n",
       "3   Probable bifunctional penicillin-binding protein   \n",
       "4                         Catalase-peroxidase enzyme   \n",
       "8                               Ribosomal protein S1   \n",
       "11                                      DNA gyrase A   \n",
       "\n",
       "   Mechanism of drug resistance  \n",
       "2                           NaN  \n",
       "3                       Unknown  \n",
       "4     Decreased drug activation  \n",
       "8           Target modification  \n",
       "11          Target modification  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resistance_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'Resistance genes'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_18268\\1020620556.py\u001b[0m in \u001b[0;36m?\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mmerged_resistance_genes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mresistance_genes\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgene_frequency\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mon\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'Resistance genes'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'left'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\core\\frame.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[0;32m  10486\u001b[0m         \u001b[0mvalidate\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mMergeValidate\u001b[0m \u001b[1;33m|\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m  10487\u001b[0m     ) -> DataFrame:\n\u001b[0;32m  10488\u001b[0m         \u001b[1;32mfrom\u001b[0m \u001b[0mpandas\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreshape\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmerge\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mmerge\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m  10489\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m> 10490\u001b[1;33m         return merge(\n\u001b[0m\u001b[0;32m  10491\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m  10492\u001b[0m             \u001b[0mright\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m  10493\u001b[0m             \u001b[0mhow\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mhow\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\core\\reshape\\merge.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[0;32m    165\u001b[0m             \u001b[0mvalidate\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mvalidate\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    166\u001b[0m             \u001b[0mcopy\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mcopy\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    167\u001b[0m         )\n\u001b[0;32m    168\u001b[0m     \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 169\u001b[1;33m         op = _MergeOperation(\n\u001b[0m\u001b[0;32m    170\u001b[0m             \u001b[0mleft_df\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    171\u001b[0m             \u001b[0mright_df\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    172\u001b[0m             \u001b[0mhow\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mhow\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\core\\reshape\\merge.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self, left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, indicator, validate)\u001b[0m\n\u001b[0;32m    787\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mright_join_keys\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    788\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjoin_names\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    789\u001b[0m             \u001b[0mleft_drop\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    790\u001b[0m             \u001b[0mright_drop\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 791\u001b[1;33m         ) = self._get_merge_keys()\n\u001b[0m\u001b[0;32m    792\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    793\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mleft_drop\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    794\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mleft\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mleft\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_drop_labels_or_levels\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mleft_drop\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\core\\reshape\\merge.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m   1265\u001b[0m                         \u001b[1;31m# Then we're either Hashable or a wrong-length arraylike,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1266\u001b[0m                         \u001b[1;31m#  the latter of which will raise\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1267\u001b[0m                         \u001b[0mrk\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcast\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mHashable\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mrk\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1268\u001b[0m                         \u001b[1;32mif\u001b[0m \u001b[0mrk\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1269\u001b[1;33m                             \u001b[0mright_keys\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mright\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_label_or_level_values\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrk\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1270\u001b[0m                         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1271\u001b[0m                             \u001b[1;31m# work-around for merge_asof(right_index=True)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1272\u001b[0m                             \u001b[0mright_keys\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mright\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self, key, axis)\u001b[0m\n\u001b[0;32m   1840\u001b[0m             \u001b[0mvalues\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mxs\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mother_axes\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1841\u001b[0m         \u001b[1;32melif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_is_level_reference\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1842\u001b[0m             \u001b[0mvalues\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0maxes\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_level_values\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1843\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1844\u001b[1;33m             \u001b[1;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1845\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1846\u001b[0m         \u001b[1;31m# Check for duplicates\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1847\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mvalues\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mndim\u001b[0m \u001b[1;33m>\u001b[0m \u001b[1;36m1\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: 'Resistance genes'"
     ]
    }
   ],
   "source": [
    "merged_resistance_genes = resistance_genes.merge(gene_frequency, on='Resistance genes', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>WHO category</th>\n",
       "      <th>Drug or drug class</th>\n",
       "      <th>Resistance genes</th>\n",
       "      <th>Rv number</th>\n",
       "      <th>Gene function</th>\n",
       "      <th>Mechanism of drug resistance</th>\n",
       "      <th>Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ponA1</td>\n",
       "      <td>Rv0050</td>\n",
       "      <td>Probable bifunctional penicillin-binding protein</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Isoniazid</td>\n",
       "      <td>katG</td>\n",
       "      <td>Rv1908c</td>\n",
       "      <td>Catalase-peroxidase enzyme</td>\n",
       "      <td>Decreased drug activation</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpsA</td>\n",
       "      <td>Rv1630</td>\n",
       "      <td>Ribosomal protein S1</td>\n",
       "      <td>Target modification</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Group A</td>\n",
       "      <td>Levofloxacin</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>Rv0006</td>\n",
       "      <td>DNA gyrase A</td>\n",
       "      <td>Target modification</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  WHO category Drug or drug class Resistance genes Rv number  \\\n",
       "0          NaN                NaN             rpoC       NaN   \n",
       "1          NaN                NaN            ponA1    Rv0050   \n",
       "2          NaN          Isoniazid             katG   Rv1908c   \n",
       "3          NaN                NaN             rpsA    Rv1630   \n",
       "4      Group A       Levofloxacin             gyrA    Rv0006   \n",
       "\n",
       "                                      Gene function  \\\n",
       "0                                               NaN   \n",
       "1  Probable bifunctional penicillin-binding protein   \n",
       "2                        Catalase-peroxidase enzyme   \n",
       "3                              Ribosomal protein S1   \n",
       "4                                      DNA gyrase A   \n",
       "\n",
       "  Mechanism of drug resistance  Frequency  \n",
       "0                          NaN          2  \n",
       "1                      Unknown          1  \n",
       "2    Decreased drug activation          3  \n",
       "3          Target modification          2  \n",
       "4          Target modification          2  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_resistance_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(11, 7)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_resistance_genes.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_resistance_genes.to_csv(\"High_CD4_results/Drug_resistance_genes_frequency.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "virulence = pd.read_csv(\"Category_virulence.csv\", sep= \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Virulence category</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ESX-1 secretion-associated protein</td>\n",
       "      <td>secretes antigens that lead to the phagosomal ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ESX-2 secretion system protein</td>\n",
       "      <td>responsible for survivability</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ESX-3 secretion system protein</td>\n",
       "      <td>involved in iron and zinc uptake and hindering...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ESX-4 secretion system protein</td>\n",
       "      <td>required in conjugation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ESX-5 type VII secretion system protein</td>\n",
       "      <td>releases PE/PPE and PE-PGRS proteins, which pl...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Virulence category  \\\n",
       "0       ESX-1 secretion-associated protein   \n",
       "1           ESX-2 secretion system protein   \n",
       "2           ESX-3 secretion system protein   \n",
       "3           ESX-4 secretion system protein   \n",
       "4  ESX-5 type VII secretion system protein   \n",
       "\n",
       "                                            Function  \n",
       "0  secretes antigens that lead to the phagosomal ...  \n",
       "1                      responsible for survivability  \n",
       "2  involved in iron and zinc uptake and hindering...  \n",
       "3                            required in conjugation  \n",
       "4  releases PE/PPE and PE-PGRS proteins, which pl...  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "virulence.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\840 G3\\AppData\\Local\\Temp\\ipykernel_18268\\2500223771.py:5: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  subset_df = low[low['Product'].str.contains('|'.join(virulence['Virulence category']))]\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Cannot mask with non-boolean array containing NA / NaN values",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32md:\\Thesis_results\\scripts\\gene_info-high.ipynb Cell 29\u001b[0m line \u001b[0;36m5\n\u001b[0;32m      <a href='vscode-notebook-cell:/d%3A/Thesis_results/scripts/gene_info-high.ipynb#Y105sZmlsZQ%3D%3D?line=1'>2</a>\u001b[0m virulence \u001b[39m=\u001b[39m virulence\u001b[39m.\u001b[39mdropna(subset\u001b[39m=\u001b[39m[\u001b[39m'\u001b[39m\u001b[39mVirulence category\u001b[39m\u001b[39m'\u001b[39m])\n\u001b[0;32m      <a href='vscode-notebook-cell:/d%3A/Thesis_results/scripts/gene_info-high.ipynb#Y105sZmlsZQ%3D%3D?line=3'>4</a>\u001b[0m \u001b[39m# Filter genes that contain any of the names in Virulence_genes\u001b[39;00m\n\u001b[1;32m----> <a href='vscode-notebook-cell:/d%3A/Thesis_results/scripts/gene_info-high.ipynb#Y105sZmlsZQ%3D%3D?line=4'>5</a>\u001b[0m subset_df \u001b[39m=\u001b[39m low[low[\u001b[39m'\u001b[39;49m\u001b[39mProduct\u001b[39;49m\u001b[39m'\u001b[39;49m]\u001b[39m.\u001b[39;49mstr\u001b[39m.\u001b[39;49mcontains(\u001b[39m'\u001b[39;49m\u001b[39m|\u001b[39;49m\u001b[39m'\u001b[39;49m\u001b[39m.\u001b[39;49mjoin(virulence[\u001b[39m'\u001b[39;49m\u001b[39mVirulence category\u001b[39;49m\u001b[39m'\u001b[39;49m]))]\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\core\\frame.py:3886\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m   3883\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mwhere(key)\n\u001b[0;32m   3885\u001b[0m \u001b[39m# Do we have a (boolean) 1d indexer?\u001b[39;00m\n\u001b[1;32m-> 3886\u001b[0m \u001b[39mif\u001b[39;00m com\u001b[39m.\u001b[39;49mis_bool_indexer(key):\n\u001b[0;32m   3887\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_getitem_bool_array(key)\n\u001b[0;32m   3889\u001b[0m \u001b[39m# We are left with two options: a single key, and a collection of keys,\u001b[39;00m\n\u001b[0;32m   3890\u001b[0m \u001b[39m# We interpret tuples as collections only for non-MultiIndex\u001b[39;00m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\core\\common.py:133\u001b[0m, in \u001b[0;36mis_bool_indexer\u001b[1;34m(key)\u001b[0m\n\u001b[0;32m    129\u001b[0m     na_msg \u001b[39m=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39mCannot mask with non-boolean array containing NA / NaN values\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[0;32m    130\u001b[0m     \u001b[39mif\u001b[39;00m lib\u001b[39m.\u001b[39mis_bool_array(key_array, skipna\u001b[39m=\u001b[39m\u001b[39mTrue\u001b[39;00m):\n\u001b[0;32m    131\u001b[0m         \u001b[39m# Don't raise on e.g. [\"A\", \"B\", np.nan], see\u001b[39;00m\n\u001b[0;32m    132\u001b[0m         \u001b[39m#  test_loc_getitem_list_of_labels_categoricalindex_with_na\u001b[39;00m\n\u001b[1;32m--> 133\u001b[0m         \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(na_msg)\n\u001b[0;32m    134\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mFalse\u001b[39;00m\n\u001b[0;32m    135\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mTrue\u001b[39;00m\n",
      "\u001b[1;31mValueError\u001b[0m: Cannot mask with non-boolean array containing NA / NaN values"
     ]
    }
   ],
   "source": [
    "# Drop rows with missing values in 'Virulence category' column\n",
    "virulence = virulence.dropna(subset=['Virulence category'])\n",
    "\n",
    "# Filter genes that contain any of the names in Virulence_genes\n",
    "subset_df = low[low['Product'].str.contains('|'.join(virulence['Virulence category']))]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "subset_data = low[low['Product'].isin(virulence['Virulence category'])].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>Effect</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Product</th>\n",
       "      <th>Gene_mutation</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7535067</th>\n",
       "      <th>SRR7535083</th>\n",
       "      <th>SRR7535084</th>\n",
       "      <th>SRR7535087</th>\n",
       "      <th>SRR7535088</th>\n",
       "      <th>SRR7535089</th>\n",
       "      <th>SRR7535090</th>\n",
       "      <th>SRR7535092</th>\n",
       "      <th>SRR7535093</th>\n",
       "      <th>Frequency_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1778</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv0001-Rv0002</td>\n",
       "      <td>dnaA-dnaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>n.1778C&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1908</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv0001-Rv0002</td>\n",
       "      <td>dnaA-dnaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>n.1908A&gt;C</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>27518</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv0022c-Rv0023</td>\n",
       "      <td>Rv0022c-Rv0022c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>n.27519delA</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>31834</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv0028-Rv0029</td>\n",
       "      <td>Rv0028-Rv0028</td>\n",
       "      <td>NaN</td>\n",
       "      <td>n.31834C&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>33202</td>\n",
       "      <td>TG</td>\n",
       "      <td>T,TGG</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv0029-Rv0030</td>\n",
       "      <td>Rv0029-Rv0029</td>\n",
       "      <td>NaN</td>\n",
       "      <td>n.33203delG</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM    POS REF    ALT       Variant_type    Effect        Locus_id  \\\n",
       "1   NC_000962   1778   C      T  intergenic_region  MODIFIER   Rv0001-Rv0002   \n",
       "2   NC_000962   1908   A      C  intergenic_region  MODIFIER   Rv0001-Rv0002   \n",
       "21  NC_000962  27518  TA      T  intergenic_region  MODIFIER  Rv0022c-Rv0023   \n",
       "23  NC_000962  31834   C      T  intergenic_region  MODIFIER   Rv0028-Rv0029   \n",
       "25  NC_000962  33202  TG  T,TGG  intergenic_region  MODIFIER   Rv0029-Rv0030   \n",
       "\n",
       "            Gene_id Product Gene_mutation  ... SRR7535067 SRR7535083  \\\n",
       "1         dnaA-dnaN     NaN     n.1778C>T  ...          0          0   \n",
       "2         dnaA-dnaN     NaN     n.1908A>C  ...          0          0   \n",
       "21  Rv0022c-Rv0022c     NaN   n.27519delA  ...          0          0   \n",
       "23    Rv0028-Rv0028     NaN    n.31834C>T  ...          0          0   \n",
       "25    Rv0029-Rv0029     NaN   n.33203delG  ...          0          0   \n",
       "\n",
       "    SRR7535084  SRR7535087  SRR7535088  SRR7535089  SRR7535090  SRR7535092  \\\n",
       "1            0           0           0           0           0           0   \n",
       "2            0           1           0           0           0           0   \n",
       "21           0           0           0           0           0           0   \n",
       "23           0           0           0           0           0           0   \n",
       "25           0           0           0           0           0           0   \n",
       "\n",
       "    SRR7535093  Frequency_Count  \n",
       "1            1                1  \n",
       "2            0                1  \n",
       "21           0                1  \n",
       "23           0                1  \n",
       "25           0                2  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>Effect</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Product</th>\n",
       "      <th>Gene_mutation</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7535067</th>\n",
       "      <th>SRR7535083</th>\n",
       "      <th>SRR7535084</th>\n",
       "      <th>SRR7535087</th>\n",
       "      <th>SRR7535088</th>\n",
       "      <th>SRR7535089</th>\n",
       "      <th>SRR7535090</th>\n",
       "      <th>SRR7535092</th>\n",
       "      <th>SRR7535093</th>\n",
       "      <th>Frequency_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>412</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0001</td>\n",
       "      <td>dnaA</td>\n",
       "      <td>chromosomal replication initiator protein DnaA</td>\n",
       "      <td>c.412G&gt;A</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1778</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv0001-Rv0002</td>\n",
       "      <td>dnaA-dnaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>n.1778C&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1908</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv0001-Rv0002</td>\n",
       "      <td>dnaA-dnaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>n.1908A&gt;C</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2883</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0002</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>DNA polymerase III subunit beta</td>\n",
       "      <td>c.832G&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>4572</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0004</td>\n",
       "      <td>Rv0004</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>c.139C&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM   POS REF ALT       Variant_type    Effect       Locus_id  \\\n",
       "0  NC_000962   412   G   A   missense_variant  MODERATE         Rv0001   \n",
       "1  NC_000962  1778   C   T  intergenic_region  MODIFIER  Rv0001-Rv0002   \n",
       "2  NC_000962  1908   A   C  intergenic_region  MODIFIER  Rv0001-Rv0002   \n",
       "3  NC_000962  2883   G   T   missense_variant  MODERATE         Rv0002   \n",
       "4  NC_000962  4572   C   T   missense_variant  MODERATE         Rv0004   \n",
       "\n",
       "     Gene_id                                         Product Gene_mutation  \\\n",
       "0       dnaA  chromosomal replication initiator protein DnaA      c.412G>A   \n",
       "1  dnaA-dnaN                                             NaN     n.1778C>T   \n",
       "2  dnaA-dnaN                                             NaN     n.1908A>C   \n",
       "3       dnaN                 DNA polymerase III subunit beta      c.832G>T   \n",
       "4     Rv0004                            hypothetical protein      c.139C>T   \n",
       "\n",
       "   ... SRR7535067 SRR7535083  SRR7535084  SRR7535087  SRR7535088  SRR7535089  \\\n",
       "0  ...          0          0           0           0           0           0   \n",
       "1  ...          0          0           0           0           0           0   \n",
       "2  ...          0          0           0           1           0           0   \n",
       "3  ...          0          0           0           1           0           0   \n",
       "4  ...          0          0           0           0           0           0   \n",
       "\n",
       "   SRR7535090  SRR7535092  SRR7535093  Frequency_Count  \n",
       "0           0           0           0                1  \n",
       "1           0           0           1                1  \n",
       "2           0           0           0                1  \n",
       "3           0           0           0                1  \n",
       "4           0           0           0                1  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "low.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_frequency = low['Gene_id'].value_counts().reset_index()\n",
    "gene_frequency.columns = ['genes', 'Frequency']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "ename": "ParserError",
     "evalue": "Error tokenizing data. C error: Expected 1 fields in line 6, saw 2\n",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mParserError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[1;32md:\\Thesis_results\\scripts\\gene_info-high.ipynb Cell 31\u001b[0m line \u001b[0;36m1\n\u001b[1;32m----> <a href='vscode-notebook-cell:/d%3A/Thesis_results/scripts/gene_info-high.ipynb#X41sZmlsZQ%3D%3D?line=0'>1</a>\u001b[0m Virulence \u001b[39m=\u001b[39m pd\u001b[39m.\u001b[39;49mread_csv(\u001b[39m\"\u001b[39;49m\u001b[39mCategory_virulence.csv\u001b[39;49m\u001b[39m\"\u001b[39;49m)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\io\\parsers\\readers.py:948\u001b[0m, in \u001b[0;36mread_csv\u001b[1;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[0;32m    935\u001b[0m kwds_defaults \u001b[39m=\u001b[39m _refine_defaults_read(\n\u001b[0;32m    936\u001b[0m     dialect,\n\u001b[0;32m    937\u001b[0m     delimiter,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    944\u001b[0m     dtype_backend\u001b[39m=\u001b[39mdtype_backend,\n\u001b[0;32m    945\u001b[0m )\n\u001b[0;32m    946\u001b[0m kwds\u001b[39m.\u001b[39mupdate(kwds_defaults)\n\u001b[1;32m--> 948\u001b[0m \u001b[39mreturn\u001b[39;00m _read(filepath_or_buffer, kwds)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\io\\parsers\\readers.py:617\u001b[0m, in \u001b[0;36m_read\u001b[1;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[0;32m    614\u001b[0m     \u001b[39mreturn\u001b[39;00m parser\n\u001b[0;32m    616\u001b[0m \u001b[39mwith\u001b[39;00m parser:\n\u001b[1;32m--> 617\u001b[0m     \u001b[39mreturn\u001b[39;00m parser\u001b[39m.\u001b[39;49mread(nrows)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\io\\parsers\\readers.py:1748\u001b[0m, in \u001b[0;36mTextFileReader.read\u001b[1;34m(self, nrows)\u001b[0m\n\u001b[0;32m   1741\u001b[0m nrows \u001b[39m=\u001b[39m validate_integer(\u001b[39m\"\u001b[39m\u001b[39mnrows\u001b[39m\u001b[39m\"\u001b[39m, nrows)\n\u001b[0;32m   1742\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m   1743\u001b[0m     \u001b[39m# error: \"ParserBase\" has no attribute \"read\"\u001b[39;00m\n\u001b[0;32m   1744\u001b[0m     (\n\u001b[0;32m   1745\u001b[0m         index,\n\u001b[0;32m   1746\u001b[0m         columns,\n\u001b[0;32m   1747\u001b[0m         col_dict,\n\u001b[1;32m-> 1748\u001b[0m     ) \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_engine\u001b[39m.\u001b[39;49mread(  \u001b[39m# type: ignore[attr-defined]\u001b[39;49;00m\n\u001b[0;32m   1749\u001b[0m         nrows\n\u001b[0;32m   1750\u001b[0m     )\n\u001b[0;32m   1751\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mException\u001b[39;00m:\n\u001b[0;32m   1752\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mclose()\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\io\\parsers\\c_parser_wrapper.py:234\u001b[0m, in \u001b[0;36mCParserWrapper.read\u001b[1;34m(self, nrows)\u001b[0m\n\u001b[0;32m    232\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m    233\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mlow_memory:\n\u001b[1;32m--> 234\u001b[0m         chunks \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_reader\u001b[39m.\u001b[39;49mread_low_memory(nrows)\n\u001b[0;32m    235\u001b[0m         \u001b[39m# destructive to chunks\u001b[39;00m\n\u001b[0;32m    236\u001b[0m         data \u001b[39m=\u001b[39m _concatenate_chunks(chunks)\n",
      "File \u001b[1;32mparsers.pyx:843\u001b[0m, in \u001b[0;36mpandas._libs.parsers.TextReader.read_low_memory\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mparsers.pyx:904\u001b[0m, in \u001b[0;36mpandas._libs.parsers.TextReader._read_rows\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mparsers.pyx:879\u001b[0m, in \u001b[0;36mpandas._libs.parsers.TextReader._tokenize_rows\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mparsers.pyx:890\u001b[0m, in \u001b[0;36mpandas._libs.parsers.TextReader._check_tokenize_status\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mparsers.pyx:2058\u001b[0m, in \u001b[0;36mpandas._libs.parsers.raise_parser_error\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mParserError\u001b[0m: Error tokenizing data. C error: Expected 1 fields in line 6, saw 2\n"
     ]
    }
   ],
   "source": [
    "Virulence = pd.read_csv(\"Category_virulence.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Virulence category</th>\n",
       "      <th>Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ESX-1 secretion-associated protein</td>\n",
       "      <td>secretes antigens that lead to the phagosomal ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ESX-2 secretion system protein</td>\n",
       "      <td>responsible for survivability</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ESX-3 secretion system protein</td>\n",
       "      <td>involved in iron and zinc uptake and hindering...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ESX-4 secretion system protein</td>\n",
       "      <td>required in conjugation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ESX-5 type VII secretion system protein</td>\n",
       "      <td>releases PE/PPE and PE-PGRS proteins, which pl...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Virulence category  \\\n",
       "0       ESX-1 secretion-associated protein   \n",
       "1           ESX-2 secretion system protein   \n",
       "2           ESX-3 secretion system protein   \n",
       "3           ESX-4 secretion system protein   \n",
       "4  ESX-5 type VII secretion system protein   \n",
       "\n",
       "                                            Function  \n",
       "0  secretes antigens that lead to the phagosomal ...  \n",
       "1                      responsible for survivability  \n",
       "2  involved in iron and zinc uptake and hindering...  \n",
       "3                            required in conjugation  \n",
       "4  releases PE/PPE and PE-PGRS proteins, which pl...  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "    Virulence.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Virulence Category</th>\n",
       "      <th>gene</th>\n",
       "      <th>Rv number</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Two component system (TCS)</td>\n",
       "      <td>dosT</td>\n",
       "      <td>Rv2027c</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Secretion system</td>\n",
       "      <td>espA</td>\n",
       "      <td>Rv3614</td>\n",
       "      <td>Esx-1 C or S</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Virulence Category  gene Rv number   Description\n",
       "7   Two component system (TCS)  dosT   Rv2027c           NaN\n",
       "15            Secretion system  espA    Rv3614  Esx-1 C or S"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "virulence_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Cannot mask with non-boolean array containing NA / NaN values",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32md:\\Thesis_results\\scripts\\gene_info-high.ipynb Cell 38\u001b[0m line \u001b[0;36m1\n\u001b[0;32m      <a href='vscode-notebook-cell:/d%3A/Thesis_results/scripts/gene_info-high.ipynb#Y106sZmlsZQ%3D%3D?line=6'>7</a>\u001b[0m category \u001b[39m=\u001b[39m row[\u001b[39m'\u001b[39m\u001b[39mVirulence category\u001b[39m\u001b[39m'\u001b[39m]\n\u001b[0;32m      <a href='vscode-notebook-cell:/d%3A/Thesis_results/scripts/gene_info-high.ipynb#Y106sZmlsZQ%3D%3D?line=8'>9</a>\u001b[0m \u001b[39m# Filter df2 for rows where 'product' starts with 'category'\u001b[39;00m\n\u001b[1;32m---> <a href='vscode-notebook-cell:/d%3A/Thesis_results/scripts/gene_info-high.ipynb#Y106sZmlsZQ%3D%3D?line=9'>10</a>\u001b[0m matching_rows \u001b[39m=\u001b[39m low[low[\u001b[39m'\u001b[39;49m\u001b[39mProduct\u001b[39;49m\u001b[39m'\u001b[39;49m]\u001b[39m.\u001b[39;49mstr\u001b[39m.\u001b[39;49mstartswith(category)]\n\u001b[0;32m     <a href='vscode-notebook-cell:/d%3A/Thesis_results/scripts/gene_info-high.ipynb#Y106sZmlsZQ%3D%3D?line=11'>12</a>\u001b[0m \u001b[39m# Get the frequency count of matching products\u001b[39;00m\n\u001b[0;32m     <a href='vscode-notebook-cell:/d%3A/Thesis_results/scripts/gene_info-high.ipynb#Y106sZmlsZQ%3D%3D?line=12'>13</a>\u001b[0m frequency_count \u001b[39m=\u001b[39m \u001b[39mlen\u001b[39m(matching_rows)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\core\\frame.py:3886\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m   3883\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mwhere(key)\n\u001b[0;32m   3885\u001b[0m \u001b[39m# Do we have a (boolean) 1d indexer?\u001b[39;00m\n\u001b[1;32m-> 3886\u001b[0m \u001b[39mif\u001b[39;00m com\u001b[39m.\u001b[39;49mis_bool_indexer(key):\n\u001b[0;32m   3887\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_getitem_bool_array(key)\n\u001b[0;32m   3889\u001b[0m \u001b[39m# We are left with two options: a single key, and a collection of keys,\u001b[39;00m\n\u001b[0;32m   3890\u001b[0m \u001b[39m# We interpret tuples as collections only for non-MultiIndex\u001b[39;00m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\core\\common.py:133\u001b[0m, in \u001b[0;36mis_bool_indexer\u001b[1;34m(key)\u001b[0m\n\u001b[0;32m    129\u001b[0m     na_msg \u001b[39m=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39mCannot mask with non-boolean array containing NA / NaN values\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[0;32m    130\u001b[0m     \u001b[39mif\u001b[39;00m lib\u001b[39m.\u001b[39mis_bool_array(key_array, skipna\u001b[39m=\u001b[39m\u001b[39mTrue\u001b[39;00m):\n\u001b[0;32m    131\u001b[0m         \u001b[39m# Don't raise on e.g. [\"A\", \"B\", np.nan], see\u001b[39;00m\n\u001b[0;32m    132\u001b[0m         \u001b[39m#  test_loc_getitem_list_of_labels_categoricalindex_with_na\u001b[39;00m\n\u001b[1;32m--> 133\u001b[0m         \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(na_msg)\n\u001b[0;32m    134\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mFalse\u001b[39;00m\n\u001b[0;32m    135\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mTrue\u001b[39;00m\n",
      "\u001b[1;31mValueError\u001b[0m: Cannot mask with non-boolean array containing NA / NaN values"
     ]
    }
   ],
   "source": [
    "# Assuming df1 has columns 'drug', 'category', and 'function', and df2 has columns 'gene' and 'product'\n",
    "# Initialize an empty list to store the frequency counts\n",
    "frequency_counts = []\n",
    "\n",
    "# Loop through each row in df1\n",
    "for index, row in virulence.iterrows():\n",
    "    category = row['Virulence category']\n",
    "    \n",
    "    # Filter df2 for rows where 'product' starts with 'category'\n",
    "    matching_rows = low[low['Product'].str.startswith(category)]\n",
    "    \n",
    "    # Get the frequency count of matching products\n",
    "    frequency_count = len(matching_rows)\n",
    "    \n",
    "    # Append the frequency count to the list\n",
    "    frequency_counts.append(frequency_count)\n",
    "\n",
    "# Add the frequency counts to df1 as a new column\n",
    "virulence['frequency_count'] = frequency_counts\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the list of drug resistance genes\n",
    "efflux_pumps = ['Rv1258c','Rv0194','emrB', 'Rv2025c',\n",
    "                'Rv3041c', 'mkl', 'Rv1463', 'Rv2326c', 'Rv3197'\n",
    "                'Rv1747','Rv0986', 'Rv0987','Rv2687c', 'Rv1456c', 'Rv2687c'\n",
    "                'drrB','dppD','dppA','Rv1273c'\n",
    "                'subI','cysW'\n",
    "]\n",
    "# Assuming 'cds_data' is your DataFrame\n",
    "\n",
    "# Subset the DataFrame based on the known list of genes\n",
    "subset_data = low[low['Gene_id'].isin(efflux_pumps)].copy()\n",
    "\n",
    "# Add a new column 'Category' indicating 'Drug Resistance'\n",
    "subset_data['Category'] = 'Efflux_pump'\n",
    "\n",
    "# Reset index if needed\n",
    "subset_data.reset_index(drop=True, inplace=True)\n",
    "\n",
    "# Print the subsetted DataFrame\n",
    "subset_data.tail()\n",
    "subset_data.to_csv(\"Low_CD4_results/Efflux_pumps.genes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "subset_data['Gene_id'].value_counts().to_csv(\"Low_CD4_results/efflux_count.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Product</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7535074</th>\n",
       "      <th>SRR7535077</th>\n",
       "      <th>SRR7535079</th>\n",
       "      <th>SRR7535080</th>\n",
       "      <th>SRR7535081</th>\n",
       "      <th>SRR7535082</th>\n",
       "      <th>SRR7535085</th>\n",
       "      <th>SRR7535086</th>\n",
       "      <th>Frequency_Count</th>\n",
       "      <th>Category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>878174</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6237.83</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv0783c</td>\n",
       "      <td>emrB</td>\n",
       "      <td>multidrug resistance protein EmrB</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>Efflux_pump</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1103786</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>6126.74</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0987</td>\n",
       "      <td>Rv0987</td>\n",
       "      <td>adhesion component ABC transporter permease</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>Efflux_pump</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1105102</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>5304.41</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0987</td>\n",
       "      <td>Rv0987</td>\n",
       "      <td>adhesion component ABC transporter permease</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>Efflux_pump</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2271368</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>7077.28</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv2025c</td>\n",
       "      <td>Rv2025c</td>\n",
       "      <td>cation efflux system protein</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>Efflux_pump</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3401501</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6050.47</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv3041c</td>\n",
       "      <td>Rv3041c</td>\n",
       "      <td>ABC transporter ATP-binding protein</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>Efflux_pump</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM      POS REF ALT     QUAL        Variant_type    EFFECT Locus_id  \\\n",
       "0  NC_000962   878174   G   A  6237.83  synonymous_variant       LOW  Rv0783c   \n",
       "1  NC_000962  1103786   G   T  6126.74    missense_variant  MODERATE   Rv0987   \n",
       "2  NC_000962  1105102   A   C  5304.41    missense_variant  MODERATE   Rv0987   \n",
       "3  NC_000962  2271368   G   A  7077.28  synonymous_variant       LOW  Rv2025c   \n",
       "4  NC_000962  3401501   G   A  6050.47    missense_variant  MODERATE  Rv3041c   \n",
       "\n",
       "   Gene_id                                      Product  ... SRR7535074  \\\n",
       "0     emrB            multidrug resistance protein EmrB  ...          0   \n",
       "1   Rv0987  adhesion component ABC transporter permease  ...          0   \n",
       "2   Rv0987  adhesion component ABC transporter permease  ...          0   \n",
       "3  Rv2025c                 cation efflux system protein  ...          0   \n",
       "4  Rv3041c          ABC transporter ATP-binding protein  ...          0   \n",
       "\n",
       "  SRR7535077 SRR7535079  SRR7535080  SRR7535081  SRR7535082  SRR7535085  \\\n",
       "0          0          0           0           0           0           0   \n",
       "1          0          0           0           0           0           0   \n",
       "2          0          0           0           0           0           0   \n",
       "3          0          0           0           0           0           0   \n",
       "4          0          0           0           0           0           0   \n",
       "\n",
       "   SRR7535086  Frequency_Count     Category  \n",
       "0           0                7  Efflux_pump  \n",
       "1           0                7  Efflux_pump  \n",
       "2           0                7  Efflux_pump  \n",
       "3           0                7  Efflux_pump  \n",
       "4           0                7  Efflux_pump  \n",
       "\n",
       "[5 rows x 60 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "virulence_genes = pd.read_csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the list of drug resistance genes\n",
    "Virulence_genes = ['ecc', 'esp', 'esx','fbp', 'hbhA','hspX','icl',\n",
    "             'ideR'\t,'irtA','irtB', 'lipF','mbtA', 'btJ','mbt'\t\n",
    "             'myc','pho', 'relA','erp','mce'\n",
    "]\n",
    "\n",
    "#Filter genes that contain any of the names in Virulence_genes\n",
    "subset_df = low[low['Gene_id'].str.contains('|'.join(Virulence_genes))]\n",
    "\n",
    "subset_df['Gene_id'].value_counts().to_csv(\"High_CD4_results/virulence_gene_counts_low.csv\")\n",
    "subset_df.to_csv(\"High_CD4_results/virulence_low_genes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Product</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7535073</th>\n",
       "      <th>SRR7535074</th>\n",
       "      <th>SRR7535077</th>\n",
       "      <th>SRR7535079</th>\n",
       "      <th>SRR7535080</th>\n",
       "      <th>SRR7535081</th>\n",
       "      <th>SRR7535082</th>\n",
       "      <th>SRR7535085</th>\n",
       "      <th>SRR7535086</th>\n",
       "      <th>Frequency_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1224307</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2109.95</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv1095-Rv1096</td>\n",
       "      <td>phoH2-NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2022126</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5798.42</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv1783</td>\n",
       "      <td>eccC5</td>\n",
       "      <td>ESX-5 type VII secretion system protein EccC5</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2035731</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>4088.75</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv1797</td>\n",
       "      <td>eccE5</td>\n",
       "      <td>ESX-5 type VII secretion system protein EccE</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 59 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #CHROM      POS REF ALT     QUAL        Variant_type    EFFECT  \\\n",
       "118  NC_000962  1224307   C   G  2109.95   intergenic_region  MODIFIER   \n",
       "161  NC_000962  2022126   G   C  5798.42    missense_variant  MODERATE   \n",
       "164  NC_000962  2035731   G   C  4088.75  synonymous_variant       LOW   \n",
       "\n",
       "          Locus_id   Gene_id                                        Product  \\\n",
       "118  Rv1095-Rv1096  phoH2-NA                                            NaN   \n",
       "161         Rv1783     eccC5  ESX-5 type VII secretion system protein EccC5   \n",
       "164         Rv1797     eccE5   ESX-5 type VII secretion system protein EccE   \n",
       "\n",
       "     ... SRR7535073 SRR7535074 SRR7535077  SRR7535079  SRR7535080  SRR7535081  \\\n",
       "118  ...          0          0          0           0           0           0   \n",
       "161  ...          0          0          0           0           0           0   \n",
       "164  ...          0          0          0           0           0           0   \n",
       "\n",
       "     SRR7535082  SRR7535085  SRR7535086  Frequency_Count  \n",
       "118           0           0           0                7  \n",
       "161           0           0           0                7  \n",
       "164           0           0           0                7  \n",
       "\n",
       "[3 rows x 59 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_df.head(n=3) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "aggregated_df = subset_df.groupby('Product').agg({\n",
    "    'Gene_id': 'first',      # Assuming 'gene id' is directly related to 'product'\n",
    "    #'Locus_id': 'first',     # Assuming 'locus id' is directly related to 'product'\n",
    "    #'Variant_type': 'first', # Assuming 'variant type' is directly related to 'product'\n",
    "    #'EFFECT' : 'first',\n",
    "    #'Gene_mutation': 'first', # Assuming 'gene mutation' is directly related to 'product'\n",
    "    #'Protein_change': 'first',    # Assuming 'AA change' is directly related to 'product'\n",
    "    #'Frequency_Count': 'sum' # Assuming 'frequency count' needs to be summed up within each 'product' group\n",
    "}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "aggregated_df.to_csv(\"Low_CD4_results/grouped_product_edited.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "aggregated_df = subset_df.groupby('Product').agg({\n",
    "    'Gene_id': 'first',      # Assuming 'gene id' is directly related to 'product'\n",
    "    'Locus_id': 'first',     # Assuming 'locus id' is directly related to 'product'\n",
    "    'Variant_type': 'first', # Assuming 'variant type' is directly related to 'product'\n",
    "    'EFFECT' : 'first',\n",
    "    'Gene_mutation': 'first', # Assuming 'gene mutation' is directly related to 'product'\n",
    "    'Protein_change': 'first',    # Assuming 'AA change' is directly related to 'product'\n",
    "    'Frequency_Count': 'sum' # Assuming 'frequency count' needs to be summed up within each 'product' group\n",
    "}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "aggregated_df.to_csv(\"Low_CD4_results/grouped.csv\" )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Product</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>Gene_mutation</th>\n",
       "      <th>Protein_change</th>\n",
       "      <th>Frequency_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2|3-dihydroxybenzoate-AMP ligase</td>\n",
       "      <td>mbtA</td>\n",
       "      <td>Rv2384</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.36T&gt;G</td>\n",
       "      <td>p.Ser12Arg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ESAT-6 like protein EsxD</td>\n",
       "      <td>esxD</td>\n",
       "      <td>Rv3891c</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.106A&gt;C</td>\n",
       "      <td>p.Asn36His</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ESAT-6 like protein EsxF</td>\n",
       "      <td>esxF</td>\n",
       "      <td>Rv3905c</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.151G&gt;A</td>\n",
       "      <td>p.Gly51Ser</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ESAT-6 like protein EsxL</td>\n",
       "      <td>esxL</td>\n",
       "      <td>Rv1198</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>c.24G&gt;T</td>\n",
       "      <td>p.Gly8Gly</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ESAT-6 like protein EsxO</td>\n",
       "      <td>esxO</td>\n",
       "      <td>Rv2346c</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>c.216G&gt;A</td>\n",
       "      <td>p.Gly72Gly</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Product Gene_id Locus_id        Variant_type  \\\n",
       "0  2|3-dihydroxybenzoate-AMP ligase    mbtA   Rv2384    missense_variant   \n",
       "1          ESAT-6 like protein EsxD    esxD  Rv3891c    missense_variant   \n",
       "2          ESAT-6 like protein EsxF    esxF  Rv3905c    missense_variant   \n",
       "3          ESAT-6 like protein EsxL    esxL   Rv1198  synonymous_variant   \n",
       "4          ESAT-6 like protein EsxO    esxO  Rv2346c  synonymous_variant   \n",
       "\n",
       "     EFFECT Gene_mutation Protein_change  Frequency_Count  \n",
       "0  MODERATE       c.36T>G     p.Ser12Arg                1  \n",
       "1  MODERATE      c.106A>C     p.Asn36His                8  \n",
       "2  MODERATE      c.151G>A     p.Gly51Ser                3  \n",
       "3       LOW       c.24G>T      p.Gly8Gly                4  \n",
       "4       LOW      c.216G>A     p.Gly72Gly                8  "
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aggregated_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Product</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7535073</th>\n",
       "      <th>SRR7535074</th>\n",
       "      <th>SRR7535077</th>\n",
       "      <th>SRR7535079</th>\n",
       "      <th>SRR7535080</th>\n",
       "      <th>SRR7535081</th>\n",
       "      <th>SRR7535082</th>\n",
       "      <th>SRR7535085</th>\n",
       "      <th>SRR7535086</th>\n",
       "      <th>Frequency_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1224307</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2109.95</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv1095-Rv1096</td>\n",
       "      <td>phoH2-NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2022126</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5798.42</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv1783</td>\n",
       "      <td>eccC5</td>\n",
       "      <td>ESX-5 type VII secretion system protein EccC5</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2035731</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>4088.75</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv1797</td>\n",
       "      <td>eccE5</td>\n",
       "      <td>ESX-5 type VII secretion system protein EccE</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3023490</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2356.14</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv2710-Rv2711</td>\n",
       "      <td>sigB-ideR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3376685</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4214.02</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv3017c</td>\n",
       "      <td>esxQ</td>\n",
       "      <td>ESAT-6 like protein EsxQ</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 59 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #CHROM      POS REF ALT     QUAL        Variant_type    EFFECT  \\\n",
       "118  NC_000962  1224307   C   G  2109.95   intergenic_region  MODIFIER   \n",
       "161  NC_000962  2022126   G   C  5798.42    missense_variant  MODERATE   \n",
       "164  NC_000962  2035731   G   C  4088.75  synonymous_variant       LOW   \n",
       "221  NC_000962  3023490   T   C  2356.14   intergenic_region  MODIFIER   \n",
       "247  NC_000962  3376685   G   A  4214.02  synonymous_variant       LOW   \n",
       "\n",
       "          Locus_id    Gene_id                                        Product  \\\n",
       "118  Rv1095-Rv1096   phoH2-NA                                            NaN   \n",
       "161         Rv1783      eccC5  ESX-5 type VII secretion system protein EccC5   \n",
       "164         Rv1797      eccE5   ESX-5 type VII secretion system protein EccE   \n",
       "221  Rv2710-Rv2711  sigB-ideR                                            NaN   \n",
       "247        Rv3017c       esxQ                       ESAT-6 like protein EsxQ   \n",
       "\n",
       "     ... SRR7535073 SRR7535074 SRR7535077  SRR7535079  SRR7535080  SRR7535081  \\\n",
       "118  ...          0          0          0           0           0           0   \n",
       "161  ...          0          0          0           0           0           0   \n",
       "164  ...          0          0          0           0           0           0   \n",
       "221  ...          0          0          0           0           0           0   \n",
       "247  ...          0          0          0           0           0           0   \n",
       "\n",
       "     SRR7535082  SRR7535085  SRR7535086  Frequency_Count  \n",
       "118           0           0           0                7  \n",
       "161           0           0           0                7  \n",
       "164           0           0           0                7  \n",
       "221           0           0           0                7  \n",
       "247           0           0           0                7  \n",
       "\n",
       "[5 rows x 59 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene_id\n",
       "relA             28\n",
       "icl1             23\n",
       "eccB4             5\n",
       "phoH1             5\n",
       "eccC3             5\n",
       "eccA3             5\n",
       "eccC2             4\n",
       "eccC4             4\n",
       "esxW              3\n",
       "eccE1             3\n",
       "espK              3\n",
       "espG2             3\n",
       "phoH2             3\n",
       "eccE2             3\n",
       "espB              3\n",
       "eccA1             2\n",
       "hspX              2\n",
       "esxF              2\n",
       "esxO              2\n",
       "eccE5             2\n",
       "esxD              2\n",
       "espI              2\n",
       "esxL              2\n",
       "irtA              2\n",
       "espH              2\n",
       "espG1             2\n",
       "eccA5             1\n",
       "eccD5             1\n",
       "eccB2             1\n",
       "Rv3888c-espG2     1\n",
       "eccCa1            1\n",
       "mbtA              1\n",
       "espB-eccE1        1\n",
       "espA-NA           1\n",
       "fbpD              1\n",
       "eccCb1            1\n",
       "phoH2-NA          1\n",
       "irtB              1\n",
       "eccA2             1\n",
       "eccD1-espJ        1\n",
       "esxB              1\n",
       "eccC5             1\n",
       "espG3             1\n",
       "eccD2             1\n",
       "Rv0128-fbpC       1\n",
       "esxP              1\n",
       "espA              1\n",
       "esxQ              1\n",
       "sigB-ideR         1\n",
       "esxF-Rv3905c      1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_df['Gene_id'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                GENE  freq_count\n",
      "0           TYPE=snp        3848\n",
      "1       TYPE=complex         662\n",
      "2           TYPE=del         114\n",
      "3           TYPE=ins          71\n",
      "4           TYPE=mnp          35\n",
      "5       TYPE=del,ins           3\n",
      "6       TYPE=snp,snp           2\n",
      "7       TYPE=ins,del           2\n",
      "8   TYPE=snp,complex           1\n",
      "9       TYPE=del,del           1\n",
      "10  TYPE=ins,ins,del           1\n",
      "11      TYPE=ins,ins           1\n",
      "12  TYPE=complex,snp           1\n",
      "13          Type=snp           1\n"
     ]
    }
   ],
   "source": [
    "# Count all unique values in the 'EFFECT' column\n",
    "gene_counts = low['TYPE'].value_counts()\n",
    "\n",
    "# Create a DataFrame with 'EFFECT' and 'low_count'\n",
    "gene_df = pd.DataFrame({'GENE': gene_counts.index, 'freq_count': gene_counts.values})\n",
    "\n",
    "# Print the counts\n",
    "print(gene_df)\n",
    "\n",
    "# Save the DataFrame to a CSV file\n",
    "gene_df.to_csv('Low_CD4_results/impact_counts_low.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             GENE  freq_count\n",
      "0            rpoB          84\n",
      "1             rrl          70\n",
      "2            rpoC          64\n",
      "3           clpC1          57\n",
      "4            rpsA          49\n",
      "...           ...         ...\n",
      "1962         argC           1\n",
      "1963      Rv1639c           1\n",
      "1964        tesB1           1\n",
      "1965  Rv1610-trpC           1\n",
      "1966      Rv3922c           1\n",
      "\n",
      "[1967 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "# Count all unique values in the 'EFFECT' column\n",
    "gene_counts = low['Gene_id'].value_counts()\n",
    "\n",
    "# Create a DataFrame with 'EFFECT' and 'low_count'\n",
    "gene_df = pd.DataFrame({'GENE': gene_counts.index, 'freq_count': gene_counts.values})\n",
    "\n",
    "# Print the counts\n",
    "print(gene_df)\n",
    "gene_df['GENE'].value_counts().to_csv(\"Low_CD4_results/gene_counts_low.csv\")\n",
    "\n",
    "# Save the DataFrame to a CSV file\n",
    "#gene_df.to_csv('Low_CD4_results/gene_counts_low.csv', index=False)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
