{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'d:\\\\Thesis_results'"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "# Change directory\n",
    "os.chdir('d:\\\\Thesis_results')\n",
    "\n",
    "# Now the current working directory is 'd:\\\\Thesis_results'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "# Define the directory path\n",
    "directory_path = 'd:\\\\Thesis_results\\\\snps.tabs\\\\'\n",
    "\n",
    "# Create a list of file paths\n",
    "files = [os.path.join(directory_path, f) for f in os.listdir(directory_path) if f.endswith(\".tab\")]\n",
    "\n",
    "# Read and concatenate the snp.tab files\n",
    "df_list = [pd.read_csv(f, sep='\\t') for f in files]\n",
    "df = pd.concat(df_list)\n",
    "\n",
    "# Reset index if needed\n",
    "df.reset_index(drop=True, inplace=True)\n",
    "\n",
    "# Remove duplicates\n",
    "df = df.drop_duplicates()\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>EVIDENCE</th>\n",
       "      <th>FTYPE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>NT_POS</th>\n",
       "      <th>AA_POS</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>LOCUS_TAG</th>\n",
       "      <th>GENE</th>\n",
       "      <th>PRODUCT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1849</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>A:139 C:0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1977</td>\n",
       "      <td>snp</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>G:80 A:0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3446</td>\n",
       "      <td>snp</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>T:145 C:0</td>\n",
       "      <td>CDS</td>\n",
       "      <td>+</td>\n",
       "      <td>167/1158</td>\n",
       "      <td>56/385</td>\n",
       "      <td>missense_variant c.167C&gt;T p.Ala56Val</td>\n",
       "      <td>Rv0003</td>\n",
       "      <td>recF</td>\n",
       "      <td>DNA replication/repair protein RecF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>4013</td>\n",
       "      <td>snp</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C:122 T:0</td>\n",
       "      <td>CDS</td>\n",
       "      <td>+</td>\n",
       "      <td>734/1158</td>\n",
       "      <td>245/385</td>\n",
       "      <td>missense_variant c.734T&gt;C p.Ile245Thr</td>\n",
       "      <td>Rv0003</td>\n",
       "      <td>recF</td>\n",
       "      <td>DNA replication/repair protein RecF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>7362</td>\n",
       "      <td>snp</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>C:141 G:0</td>\n",
       "      <td>CDS</td>\n",
       "      <td>+</td>\n",
       "      <td>61/2517</td>\n",
       "      <td>21/838</td>\n",
       "      <td>missense_variant c.61G&gt;C p.Glu21Gln</td>\n",
       "      <td>Rv0006</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>DNA gyrase subunit A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM   POS TYPE REF ALT   EVIDENCE FTYPE STRAND    NT_POS   AA_POS  \\\n",
       "0  NC_000962  1849  snp   C   A  A:139 C:0   NaN    NaN       NaN      NaN   \n",
       "1  NC_000962  1977  snp   A   G   G:80 A:0   NaN    NaN       NaN      NaN   \n",
       "2  NC_000962  3446  snp   C   T  T:145 C:0   CDS      +  167/1158   56/385   \n",
       "3  NC_000962  4013  snp   T   C  C:122 T:0   CDS      +  734/1158  245/385   \n",
       "4  NC_000962  7362  snp   G   C  C:141 G:0   CDS      +   61/2517   21/838   \n",
       "\n",
       "                                  EFFECT LOCUS_TAG  GENE  \\\n",
       "0                                    NaN       NaN   NaN   \n",
       "1                                    NaN       NaN   NaN   \n",
       "2   missense_variant c.167C>T p.Ala56Val    Rv0003  recF   \n",
       "3  missense_variant c.734T>C p.Ile245Thr    Rv0003  recF   \n",
       "4    missense_variant c.61G>C p.Glu21Gln    Rv0006  gyrA   \n",
       "\n",
       "                               PRODUCT  \n",
       "0                                  NaN  \n",
       "1                                  NaN  \n",
       "2  DNA replication/repair protein RecF  \n",
       "3  DNA replication/repair protein RecF  \n",
       "4                 DNA gyrase subunit A  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'d:\\\\Thesis_results'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd\n",
    "low = pd.read_csv(\"LOW_CD4_merged_variants.txt\", sep='\\t')\n",
    "#low = pd.read_csv(\"HIGH_CD4_merged_variants.txt\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "low.shape#_4743 variants among low CD4 cell individuals\n",
    "#1326 variants among individuals with high and low CD4 cell count\n",
    "low['TYPE'].value_counts().to_csv(\"Low_CD4_results/SNP_TYPE.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Product</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7535073</th>\n",
       "      <th>SRR7535074</th>\n",
       "      <th>SRR7535077</th>\n",
       "      <th>SRR7535079</th>\n",
       "      <th>SRR7535080</th>\n",
       "      <th>SRR7535081</th>\n",
       "      <th>SRR7535082</th>\n",
       "      <th>SRR7535085</th>\n",
       "      <th>SRR7535086</th>\n",
       "      <th>Frequency_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>337820</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>944.650</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv0279c</td>\n",
       "      <td>PE_PGRS4</td>\n",
       "      <td>PE-PGRS family protein PE_PGRS4</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>623508</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1401.370</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0532</td>\n",
       "      <td>PE_PGRS6</td>\n",
       "      <td>PE-PGRS family protein PE_PGRS6</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>623472</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>915.797</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0532</td>\n",
       "      <td>PE_PGRS6</td>\n",
       "      <td>PE-PGRS family protein PE_PGRS6</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>580772</td>\n",
       "      <td>TGGGGGCACCACCCGCTTGCGGGGGA</td>\n",
       "      <td>AG</td>\n",
       "      <td>364.351</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv0490-Rv0491</td>\n",
       "      <td>senX3-NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3069566</td>\n",
       "      <td>C</td>\n",
       "      <td>T,A</td>\n",
       "      <td>6225.910</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv2756c</td>\n",
       "      <td>hsdM</td>\n",
       "      <td>type I restriction/modification system DNA met...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>217201</td>\n",
       "      <td>TGCGCA</td>\n",
       "      <td>CGCGCA,CGCG</td>\n",
       "      <td>5668.940</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv0186</td>\n",
       "      <td>bglS</td>\n",
       "      <td>beta-glucosidase BglS</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>836538</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>701.428</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0746</td>\n",
       "      <td>PE_PGRS9</td>\n",
       "      <td>PE-PGRS family protein PE_PGRS9</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3847099</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1045.810</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv3428c-Rv3429</td>\n",
       "      <td>Rv3428c-PPE59</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3847112</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1293.960</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv3428c-Rv3429</td>\n",
       "      <td>Rv3428c-PPE59</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>836454</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>919.422</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0746</td>\n",
       "      <td>PE_PGRS9</td>\n",
       "      <td>PE-PGRS family protein PE_PGRS9</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 59 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM      POS                         REF          ALT      QUAL  \\\n",
       "0  NC_000962   337820                           G            A   944.650   \n",
       "1  NC_000962   623508                           C            G  1401.370   \n",
       "2  NC_000962   623472                           A            G   915.797   \n",
       "3  NC_000962   580772  TGGGGGCACCACCCGCTTGCGGGGGA           AG   364.351   \n",
       "4  NC_000962  3069566                           C          T,A  6225.910   \n",
       "5  NC_000962   217201                      TGCGCA  CGCGCA,CGCG  5668.940   \n",
       "6  NC_000962   836538                           A            G   701.428   \n",
       "7  NC_000962  3847099                           G            A  1045.810   \n",
       "8  NC_000962  3847112                           T            A  1293.960   \n",
       "9  NC_000962   836454                           A            G   919.422   \n",
       "\n",
       "         Variant_type    EFFECT        Locus_id        Gene_id  \\\n",
       "0  synonymous_variant       LOW         Rv0279c       PE_PGRS4   \n",
       "1    missense_variant  MODERATE          Rv0532       PE_PGRS6   \n",
       "2    missense_variant  MODERATE          Rv0532       PE_PGRS6   \n",
       "3   intergenic_region  MODIFIER   Rv0490-Rv0491       senX3-NA   \n",
       "4    missense_variant  MODERATE         Rv2756c           hsdM   \n",
       "5  synonymous_variant       LOW          Rv0186           bglS   \n",
       "6    missense_variant  MODERATE          Rv0746       PE_PGRS9   \n",
       "7   intergenic_region  MODIFIER  Rv3428c-Rv3429  Rv3428c-PPE59   \n",
       "8   intergenic_region  MODIFIER  Rv3428c-Rv3429  Rv3428c-PPE59   \n",
       "9    missense_variant  MODERATE          Rv0746       PE_PGRS9   \n",
       "\n",
       "                                             Product  ... SRR7535073  \\\n",
       "0                    PE-PGRS family protein PE_PGRS4  ...          0   \n",
       "1                    PE-PGRS family protein PE_PGRS6  ...          0   \n",
       "2                    PE-PGRS family protein PE_PGRS6  ...          0   \n",
       "3                                                NaN  ...          0   \n",
       "4  type I restriction/modification system DNA met...  ...          0   \n",
       "5                              beta-glucosidase BglS  ...          0   \n",
       "6                    PE-PGRS family protein PE_PGRS9  ...          0   \n",
       "7                                                NaN  ...          0   \n",
       "8                                                NaN  ...          0   \n",
       "9                    PE-PGRS family protein PE_PGRS9  ...          0   \n",
       "\n",
       "  SRR7535074 SRR7535077  SRR7535079  SRR7535080  SRR7535081  SRR7535082  \\\n",
       "0          0          0           0           0           0           0   \n",
       "1          0          0           0           0           0           0   \n",
       "2          0          0           0           0           0           0   \n",
       "3          0          0           0           0           0           0   \n",
       "4          1          0           0           0           0           0   \n",
       "5          1          0           0           0           0           0   \n",
       "6          0          0           0           0           0           0   \n",
       "7          0          0           0           0           0           0   \n",
       "8          0          0           0           0           0           0   \n",
       "9          0          0           0           0           0           0   \n",
       "\n",
       "   SRR7535085  SRR7535086  Frequency_Count  \n",
       "0           0           0               28  \n",
       "1           1           0               23  \n",
       "2           1           1               20  \n",
       "3           0           0               13  \n",
       "4           0           0               13  \n",
       "5           0           0               12  \n",
       "6           0           0               12  \n",
       "7           0           0               12  \n",
       "8           0           0               12  \n",
       "9           0           0               11  \n",
       "\n",
       "[10 rows x 59 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "low.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Assuming df is your original DataFrame\n",
    "\n",
    "# Create a new DataFrame containing rows with NaN values\n",
    "intergenic = low[low.isna().any(axis=1)].copy()\n",
    "\n",
    "# Reset index if needed\n",
    "intergenic.reset_index(drop=True, inplace=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(512, 59)"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intergenic.head()\n",
    "intergenic.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         Variant_type  low_count\n",
      "0                                  synonymous_variant       2361\n",
      "1                                    missense_variant       1667\n",
      "2                                   intergenic_region        403\n",
      "3                                  frameshift_variant        107\n",
      "4                       non_coding_transcript_variant        100\n",
      "5                                         stop_gained         36\n",
      "6                         disruptive_inframe_deletion         24\n",
      "7                        disruptive_inframe_insertion         12\n",
      "8                      conservative_inframe_insertion          8\n",
      "9                       conservative_inframe_deletion          6\n",
      "10                                         start_lost          5\n",
      "11                    stop_lost&splice_region_variant          4\n",
      "12                     frameshift_variant&stop_gained          3\n",
      "13                frameshift_variant&missense_variant          3\n",
      "14     missense_variant&conservative_inframe_deletion          1\n",
      "15   conservative_inframe_deletion&synonymous_variant          1\n",
      "16                      frameshift_variant&start_lost          1\n",
      "17  disruptive_inframe_insertion&splice_region_var...          1\n"
     ]
    }
   ],
   "source": [
    "# Count all unique values in the 'EFFECT' column\n",
    "effect_counts = low['Variant_type'].value_counts()\n",
    "\n",
    "# Create a DataFrame with 'EFFECT' and 'low_count'\n",
    "effect_df_1 = pd.DataFrame({'Variant_type': effect_counts.index, 'low_count': effect_counts.values})\n",
    "\n",
    "# Print the counts\n",
    "print(effect_df_1)\n",
    "\n",
    "# Save the DataFrame to a CSV file\n",
    "effect_df_1.to_csv('Low_CD4_results/effect_count.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fill empty 'Gene' values with corresponding 'LocusID'\n",
    "low['Gene_id'].fillna(low['Locus_id'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_panel = pd.read_csv(\"resistance genes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(41, 6)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_panel.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>WHO category</th>\n",
       "      <th>Drug or drug class</th>\n",
       "      <th>Resistance genes</th>\n",
       "      <th>Rv number</th>\n",
       "      <th>Gene function</th>\n",
       "      <th>Mechanism of drug resistance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>First-line agents</td>\n",
       "      <td>Rifamycins (for example, rifampicin)</td>\n",
       "      <td>rpoB</td>\n",
       "      <td>Rv0667</td>\n",
       "      <td>RNA polymerase</td>\n",
       "      <td>Target modification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoA</td>\n",
       "      <td>Rv0667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ponA1</td>\n",
       "      <td>Rv0050</td>\n",
       "      <td>Probable bifunctional penicillin-binding protein</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Isoniazid</td>\n",
       "      <td>katG</td>\n",
       "      <td>Rv1908c</td>\n",
       "      <td>Catalase-peroxidase enzyme</td>\n",
       "      <td>Decreased drug activation</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        WHO category                    Drug or drug class Resistance genes  \\\n",
       "0  First-line agents  Rifamycins (for example, rifampicin)             rpoB   \n",
       "1                NaN                                   NaN             rpoA   \n",
       "2                NaN                                   NaN             rpoC   \n",
       "3                NaN                                   NaN            ponA1   \n",
       "4                NaN                             Isoniazid             katG   \n",
       "\n",
       "  Rv number                                     Gene function  \\\n",
       "0    Rv0667                                    RNA polymerase   \n",
       "1    Rv0667                                               NaN   \n",
       "2       NaN                                               NaN   \n",
       "3    Rv0050  Probable bifunctional penicillin-binding protein   \n",
       "4   Rv1908c                        Catalase-peroxidase enzyme   \n",
       "\n",
       "  Mechanism of drug resistance  \n",
       "0          Target modification  \n",
       "1                          NaN  \n",
       "2                          NaN  \n",
       "3                      Unknown  \n",
       "4    Decreased drug activation  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_panel.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_frequency = low['Gene_id'].value_counts().reset_index()\n",
    "gene_frequency.columns = ['Resistance genes', 'Frequency']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1967, 2)"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_frequency.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Resistance genes</th>\n",
       "      <th>Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rpoB</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rrl</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rpoC</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>clpC1</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rpsA</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Resistance genes  Frequency\n",
       "0             rpoB         84\n",
       "1              rrl         70\n",
       "2             rpoC         64\n",
       "3            clpC1         57\n",
       "4             rpsA         49"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_frequency.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "resistance_genes = gene_panel[gene_panel['Resistance genes'].isin(low['Gene_id'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(19, 6)"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resistance_genes.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>WHO category</th>\n",
       "      <th>Drug or drug class</th>\n",
       "      <th>Resistance genes</th>\n",
       "      <th>Rv number</th>\n",
       "      <th>Gene function</th>\n",
       "      <th>Mechanism of drug resistance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>First-line agents</td>\n",
       "      <td>Rifamycins (for example, rifampicin)</td>\n",
       "      <td>rpoB</td>\n",
       "      <td>Rv0667</td>\n",
       "      <td>RNA polymerase</td>\n",
       "      <td>Target modification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoA</td>\n",
       "      <td>Rv0667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ponA1</td>\n",
       "      <td>Rv0050</td>\n",
       "      <td>Probable bifunctional penicillin-binding protein</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Isoniazid</td>\n",
       "      <td>katG</td>\n",
       "      <td>Rv1908c</td>\n",
       "      <td>Catalase-peroxidase enzyme</td>\n",
       "      <td>Decreased drug activation</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        WHO category                    Drug or drug class Resistance genes  \\\n",
       "0  First-line agents  Rifamycins (for example, rifampicin)             rpoB   \n",
       "1                NaN                                   NaN             rpoA   \n",
       "2                NaN                                   NaN             rpoC   \n",
       "3                NaN                                   NaN            ponA1   \n",
       "4                NaN                             Isoniazid             katG   \n",
       "\n",
       "  Rv number                                     Gene function  \\\n",
       "0    Rv0667                                    RNA polymerase   \n",
       "1    Rv0667                                               NaN   \n",
       "2       NaN                                               NaN   \n",
       "3    Rv0050  Probable bifunctional penicillin-binding protein   \n",
       "4   Rv1908c                        Catalase-peroxidase enzyme   \n",
       "\n",
       "  Mechanism of drug resistance  \n",
       "0          Target modification  \n",
       "1                          NaN  \n",
       "2                          NaN  \n",
       "3                      Unknown  \n",
       "4    Decreased drug activation  "
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resistance_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_resistance_genes = resistance_genes.merge(gene_frequency, on='Resistance genes', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>WHO category</th>\n",
       "      <th>Drug or drug class</th>\n",
       "      <th>Resistance genes</th>\n",
       "      <th>Rv number</th>\n",
       "      <th>Gene function</th>\n",
       "      <th>Mechanism of drug resistance</th>\n",
       "      <th>Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>First-line agents</td>\n",
       "      <td>Rifamycins (for example, rifampicin)</td>\n",
       "      <td>rpoB</td>\n",
       "      <td>Rv0667</td>\n",
       "      <td>RNA polymerase</td>\n",
       "      <td>Target modification</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoA</td>\n",
       "      <td>Rv0667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpoC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ponA1</td>\n",
       "      <td>Rv0050</td>\n",
       "      <td>Probable bifunctional penicillin-binding protein</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Isoniazid</td>\n",
       "      <td>katG</td>\n",
       "      <td>Rv1908c</td>\n",
       "      <td>Catalase-peroxidase enzyme</td>\n",
       "      <td>Decreased drug activation</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        WHO category                    Drug or drug class Resistance genes  \\\n",
       "0  First-line agents  Rifamycins (for example, rifampicin)             rpoB   \n",
       "1                NaN                                   NaN             rpoA   \n",
       "2                NaN                                   NaN             rpoC   \n",
       "3                NaN                                   NaN            ponA1   \n",
       "4                NaN                             Isoniazid             katG   \n",
       "\n",
       "  Rv number                                     Gene function  \\\n",
       "0    Rv0667                                    RNA polymerase   \n",
       "1    Rv0667                                               NaN   \n",
       "2       NaN                                               NaN   \n",
       "3    Rv0050  Probable bifunctional penicillin-binding protein   \n",
       "4   Rv1908c                        Catalase-peroxidase enzyme   \n",
       "\n",
       "  Mechanism of drug resistance  Frequency  \n",
       "0          Target modification         84  \n",
       "1                          NaN         30  \n",
       "2                          NaN         64  \n",
       "3                      Unknown          1  \n",
       "4    Decreased drug activation          1  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_resistance_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(19, 7)"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_resistance_genes.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_resistance_genes.to_csv(\"Low_CD4_results/Drug_resistance_genes_frequency.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the list of drug resistance genes\n",
    "efflux_pumps = ['Rv1258c','Rv0194','emrB', 'Rv2025c',\n",
    "                'Rv3041c', 'mkl', 'Rv1463', 'Rv2326c', 'Rv3197'\n",
    "                'Rv1747','Rv0986', 'Rv0987','Rv2687c', 'Rv1456c', 'Rv2687c'\n",
    "                'drrB','dppD','dppA','Rv1273c'\n",
    "                'subI','cysW'\n",
    "]\n",
    "# Assuming 'cds_data' is your DataFrame\n",
    "\n",
    "# Subset the DataFrame based on the known list of genes\n",
    "subset_data = low[low['Gene_id'].isin(efflux_pumps)].copy()\n",
    "\n",
    "# Add a new column 'Category' indicating 'Drug Resistance'\n",
    "subset_data['Category'] = 'Efflux_pump'\n",
    "\n",
    "# Reset index if needed\n",
    "subset_data.reset_index(drop=True, inplace=True)\n",
    "\n",
    "# Print the subsetted DataFrame\n",
    "subset_data.tail()\n",
    "subset_data.to_csv(\"Low_CD4_results/Efflux_pumps.genes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "subset_data['Gene_id'].value_counts().to_csv(\"Low_CD4_results/efflux_count.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Product</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7535074</th>\n",
       "      <th>SRR7535077</th>\n",
       "      <th>SRR7535079</th>\n",
       "      <th>SRR7535080</th>\n",
       "      <th>SRR7535081</th>\n",
       "      <th>SRR7535082</th>\n",
       "      <th>SRR7535085</th>\n",
       "      <th>SRR7535086</th>\n",
       "      <th>Frequency_Count</th>\n",
       "      <th>Category</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>878174</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6237.83</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv0783c</td>\n",
       "      <td>emrB</td>\n",
       "      <td>multidrug resistance protein EmrB</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>Efflux_pump</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1103786</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>6126.74</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0987</td>\n",
       "      <td>Rv0987</td>\n",
       "      <td>adhesion component ABC transporter permease</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>Efflux_pump</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1105102</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>5304.41</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv0987</td>\n",
       "      <td>Rv0987</td>\n",
       "      <td>adhesion component ABC transporter permease</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>Efflux_pump</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2271368</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>7077.28</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv2025c</td>\n",
       "      <td>Rv2025c</td>\n",
       "      <td>cation efflux system protein</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>Efflux_pump</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3401501</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6050.47</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv3041c</td>\n",
       "      <td>Rv3041c</td>\n",
       "      <td>ABC transporter ATP-binding protein</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>Efflux_pump</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #CHROM      POS REF ALT     QUAL        Variant_type    EFFECT Locus_id  \\\n",
       "0  NC_000962   878174   G   A  6237.83  synonymous_variant       LOW  Rv0783c   \n",
       "1  NC_000962  1103786   G   T  6126.74    missense_variant  MODERATE   Rv0987   \n",
       "2  NC_000962  1105102   A   C  5304.41    missense_variant  MODERATE   Rv0987   \n",
       "3  NC_000962  2271368   G   A  7077.28  synonymous_variant       LOW  Rv2025c   \n",
       "4  NC_000962  3401501   G   A  6050.47    missense_variant  MODERATE  Rv3041c   \n",
       "\n",
       "   Gene_id                                      Product  ... SRR7535074  \\\n",
       "0     emrB            multidrug resistance protein EmrB  ...          0   \n",
       "1   Rv0987  adhesion component ABC transporter permease  ...          0   \n",
       "2   Rv0987  adhesion component ABC transporter permease  ...          0   \n",
       "3  Rv2025c                 cation efflux system protein  ...          0   \n",
       "4  Rv3041c          ABC transporter ATP-binding protein  ...          0   \n",
       "\n",
       "  SRR7535077 SRR7535079  SRR7535080  SRR7535081  SRR7535082  SRR7535085  \\\n",
       "0          0          0           0           0           0           0   \n",
       "1          0          0           0           0           0           0   \n",
       "2          0          0           0           0           0           0   \n",
       "3          0          0           0           0           0           0   \n",
       "4          0          0           0           0           0           0   \n",
       "\n",
       "   SRR7535086  Frequency_Count     Category  \n",
       "0           0                7  Efflux_pump  \n",
       "1           0                7  Efflux_pump  \n",
       "2           0                7  Efflux_pump  \n",
       "3           0                7  Efflux_pump  \n",
       "4           0                7  Efflux_pump  \n",
       "\n",
       "[5 rows x 60 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the list of drug resistance genes\n",
    "Virulence_genes = ['Rv1794','ecc', 'esp', 'esx','fbp', 'hbhA','hspX','icl',\n",
    "             'ideR'\t,'irtA','irtB', 'lipF','mbtA', 'btJ','mbt'\t\n",
    "             'myc','pho', 'relA','erp','mce'\n",
    "]\n",
    "\n",
    "#Filter genes that contain any of the names in Virulence_genes\n",
    "subset_df = low[low['Gene_id'].str.contains('|'.join(Virulence_genes))]\n",
    "\n",
    "subset_df['Gene_id'].value_counts().to_csv(\"Low_CD4_results/virulence_gene_counts_low.csv\")\n",
    "subset_df.to_csv(\"Low_CD4_results/virulence_low_genes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Product</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7535073</th>\n",
       "      <th>SRR7535074</th>\n",
       "      <th>SRR7535077</th>\n",
       "      <th>SRR7535079</th>\n",
       "      <th>SRR7535080</th>\n",
       "      <th>SRR7535081</th>\n",
       "      <th>SRR7535082</th>\n",
       "      <th>SRR7535085</th>\n",
       "      <th>SRR7535086</th>\n",
       "      <th>Frequency_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1224307</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2109.95</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv1095-Rv1096</td>\n",
       "      <td>phoH2-NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2022126</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5798.42</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv1783</td>\n",
       "      <td>eccC5</td>\n",
       "      <td>ESX-5 type VII secretion system protein EccC5</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2035731</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>4088.75</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv1797</td>\n",
       "      <td>eccE5</td>\n",
       "      <td>ESX-5 type VII secretion system protein EccE</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 59 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #CHROM      POS REF ALT     QUAL        Variant_type    EFFECT  \\\n",
       "118  NC_000962  1224307   C   G  2109.95   intergenic_region  MODIFIER   \n",
       "161  NC_000962  2022126   G   C  5798.42    missense_variant  MODERATE   \n",
       "164  NC_000962  2035731   G   C  4088.75  synonymous_variant       LOW   \n",
       "\n",
       "          Locus_id   Gene_id                                        Product  \\\n",
       "118  Rv1095-Rv1096  phoH2-NA                                            NaN   \n",
       "161         Rv1783     eccC5  ESX-5 type VII secretion system protein EccC5   \n",
       "164         Rv1797     eccE5   ESX-5 type VII secretion system protein EccE   \n",
       "\n",
       "     ... SRR7535073 SRR7535074 SRR7535077  SRR7535079  SRR7535080  SRR7535081  \\\n",
       "118  ...          0          0          0           0           0           0   \n",
       "161  ...          0          0          0           0           0           0   \n",
       "164  ...          0          0          0           0           0           0   \n",
       "\n",
       "     SRR7535082  SRR7535085  SRR7535086  Frequency_Count  \n",
       "118           0           0           0                7  \n",
       "161           0           0           0                7  \n",
       "164           0           0           0                7  \n",
       "\n",
       "[3 rows x 59 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_df.head(n=3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "aggregated_df = subset_df.groupby('Product').agg({\n",
    "    'Gene_id': 'first',      # Assuming 'gene id' is directly related to 'product'\n",
    "    #'Locus_id': 'first',     # Assuming 'locus id' is directly related to 'product'\n",
    "    #'Variant_type': 'first', # Assuming 'variant type' is directly related to 'product'\n",
    "    #'EFFECT' : 'first',\n",
    "    #'Gene_mutation': 'first', # Assuming 'gene mutation' is directly related to 'product'\n",
    "    #'Protein_change': 'first',    # Assuming 'AA change' is directly related to 'product'\n",
    "    #'Frequency_Count': 'sum' # Assuming 'frequency count' needs to be summed up within each 'product' group\n",
    "}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "aggregated_df.to_csv(\"Low_CD4_results/grouped_product_edited.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "aggregated_df = subset_df.groupby('Product').agg({\n",
    "    'Gene_id': 'first',      # Assuming 'gene id' is directly related to 'product'\n",
    "    'Locus_id': 'first',     # Assuming 'locus id' is directly related to 'product'\n",
    "    'Variant_type': 'first', # Assuming 'variant type' is directly related to 'product'\n",
    "    'EFFECT' : 'first',\n",
    "    'Gene_mutation': 'first', # Assuming 'gene mutation' is directly related to 'product'\n",
    "    'Protein_change': 'first',    # Assuming 'AA change' is directly related to 'product'\n",
    "    'Frequency_Count': 'sum' # Assuming 'frequency count' needs to be summed up within each 'product' group\n",
    "}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "aggregated_df.to_csv(\"Low_CD4_results/grouped.csv\" )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Product</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>Gene_mutation</th>\n",
       "      <th>Protein_change</th>\n",
       "      <th>Frequency_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2|3-dihydroxybenzoate-AMP ligase</td>\n",
       "      <td>mbtA</td>\n",
       "      <td>Rv2384</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.36T&gt;G</td>\n",
       "      <td>p.Ser12Arg</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ESAT-6 like protein EsxD</td>\n",
       "      <td>esxD</td>\n",
       "      <td>Rv3891c</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.106A&gt;C</td>\n",
       "      <td>p.Asn36His</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ESAT-6 like protein EsxF</td>\n",
       "      <td>esxF</td>\n",
       "      <td>Rv3905c</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>c.151G&gt;A</td>\n",
       "      <td>p.Gly51Ser</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ESAT-6 like protein EsxL</td>\n",
       "      <td>esxL</td>\n",
       "      <td>Rv1198</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>c.24G&gt;T</td>\n",
       "      <td>p.Gly8Gly</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ESAT-6 like protein EsxO</td>\n",
       "      <td>esxO</td>\n",
       "      <td>Rv2346c</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>c.216G&gt;A</td>\n",
       "      <td>p.Gly72Gly</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Product Gene_id Locus_id        Variant_type  \\\n",
       "0  2|3-dihydroxybenzoate-AMP ligase    mbtA   Rv2384    missense_variant   \n",
       "1          ESAT-6 like protein EsxD    esxD  Rv3891c    missense_variant   \n",
       "2          ESAT-6 like protein EsxF    esxF  Rv3905c    missense_variant   \n",
       "3          ESAT-6 like protein EsxL    esxL   Rv1198  synonymous_variant   \n",
       "4          ESAT-6 like protein EsxO    esxO  Rv2346c  synonymous_variant   \n",
       "\n",
       "     EFFECT Gene_mutation Protein_change  Frequency_Count  \n",
       "0  MODERATE       c.36T>G     p.Ser12Arg                1  \n",
       "1  MODERATE      c.106A>C     p.Asn36His                8  \n",
       "2  MODERATE      c.151G>A     p.Gly51Ser                3  \n",
       "3       LOW       c.24G>T      p.Gly8Gly                4  \n",
       "4       LOW      c.216G>A     p.Gly72Gly                8  "
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aggregated_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>Variant_type</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>Locus_id</th>\n",
       "      <th>Gene_id</th>\n",
       "      <th>Product</th>\n",
       "      <th>...</th>\n",
       "      <th>SRR7535073</th>\n",
       "      <th>SRR7535074</th>\n",
       "      <th>SRR7535077</th>\n",
       "      <th>SRR7535079</th>\n",
       "      <th>SRR7535080</th>\n",
       "      <th>SRR7535081</th>\n",
       "      <th>SRR7535082</th>\n",
       "      <th>SRR7535085</th>\n",
       "      <th>SRR7535086</th>\n",
       "      <th>Frequency_Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>1224307</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2109.95</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv1095-Rv1096</td>\n",
       "      <td>phoH2-NA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2022126</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5798.42</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>Rv1783</td>\n",
       "      <td>eccC5</td>\n",
       "      <td>ESX-5 type VII secretion system protein EccC5</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>2035731</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>4088.75</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv1797</td>\n",
       "      <td>eccE5</td>\n",
       "      <td>ESX-5 type VII secretion system protein EccE</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3023490</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2356.14</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rv2710-Rv2711</td>\n",
       "      <td>sigB-ideR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>NC_000962</td>\n",
       "      <td>3376685</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4214.02</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>Rv3017c</td>\n",
       "      <td>esxQ</td>\n",
       "      <td>ESAT-6 like protein EsxQ</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 59 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #CHROM      POS REF ALT     QUAL        Variant_type    EFFECT  \\\n",
       "118  NC_000962  1224307   C   G  2109.95   intergenic_region  MODIFIER   \n",
       "161  NC_000962  2022126   G   C  5798.42    missense_variant  MODERATE   \n",
       "164  NC_000962  2035731   G   C  4088.75  synonymous_variant       LOW   \n",
       "221  NC_000962  3023490   T   C  2356.14   intergenic_region  MODIFIER   \n",
       "247  NC_000962  3376685   G   A  4214.02  synonymous_variant       LOW   \n",
       "\n",
       "          Locus_id    Gene_id                                        Product  \\\n",
       "118  Rv1095-Rv1096   phoH2-NA                                            NaN   \n",
       "161         Rv1783      eccC5  ESX-5 type VII secretion system protein EccC5   \n",
       "164         Rv1797      eccE5   ESX-5 type VII secretion system protein EccE   \n",
       "221  Rv2710-Rv2711  sigB-ideR                                            NaN   \n",
       "247        Rv3017c       esxQ                       ESAT-6 like protein EsxQ   \n",
       "\n",
       "     ... SRR7535073 SRR7535074 SRR7535077  SRR7535079  SRR7535080  SRR7535081  \\\n",
       "118  ...          0          0          0           0           0           0   \n",
       "161  ...          0          0          0           0           0           0   \n",
       "164  ...          0          0          0           0           0           0   \n",
       "221  ...          0          0          0           0           0           0   \n",
       "247  ...          0          0          0           0           0           0   \n",
       "\n",
       "     SRR7535082  SRR7535085  SRR7535086  Frequency_Count  \n",
       "118           0           0           0                7  \n",
       "161           0           0           0                7  \n",
       "164           0           0           0                7  \n",
       "221           0           0           0                7  \n",
       "247           0           0           0                7  \n",
       "\n",
       "[5 rows x 59 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene_id\n",
       "relA             28\n",
       "icl1             23\n",
       "eccB4             5\n",
       "phoH1             5\n",
       "eccC3             5\n",
       "eccA3             5\n",
       "eccC2             4\n",
       "eccC4             4\n",
       "esxW              3\n",
       "eccE1             3\n",
       "espK              3\n",
       "espG2             3\n",
       "phoH2             3\n",
       "eccE2             3\n",
       "espB              3\n",
       "eccA1             2\n",
       "hspX              2\n",
       "esxF              2\n",
       "esxO              2\n",
       "eccE5             2\n",
       "esxD              2\n",
       "espI              2\n",
       "esxL              2\n",
       "irtA              2\n",
       "espH              2\n",
       "espG1             2\n",
       "eccA5             1\n",
       "eccD5             1\n",
       "eccB2             1\n",
       "Rv3888c-espG2     1\n",
       "eccCa1            1\n",
       "mbtA              1\n",
       "espB-eccE1        1\n",
       "espA-NA           1\n",
       "fbpD              1\n",
       "eccCb1            1\n",
       "phoH2-NA          1\n",
       "irtB              1\n",
       "eccA2             1\n",
       "eccD1-espJ        1\n",
       "esxB              1\n",
       "eccC5             1\n",
       "espG3             1\n",
       "eccD2             1\n",
       "Rv0128-fbpC       1\n",
       "esxP              1\n",
       "espA              1\n",
       "esxQ              1\n",
       "sigB-ideR         1\n",
       "esxF-Rv3905c      1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_df['Gene_id'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                GENE  freq_count\n",
      "0           TYPE=snp        3848\n",
      "1       TYPE=complex         662\n",
      "2           TYPE=del         114\n",
      "3           TYPE=ins          71\n",
      "4           TYPE=mnp          35\n",
      "5       TYPE=del,ins           3\n",
      "6       TYPE=snp,snp           2\n",
      "7       TYPE=ins,del           2\n",
      "8   TYPE=snp,complex           1\n",
      "9       TYPE=del,del           1\n",
      "10  TYPE=ins,ins,del           1\n",
      "11      TYPE=ins,ins           1\n",
      "12  TYPE=complex,snp           1\n",
      "13          Type=snp           1\n"
     ]
    }
   ],
   "source": [
    "# Count all unique values in the 'EFFECT' column\n",
    "gene_counts = low['TYPE'].value_counts()\n",
    "\n",
    "# Create a DataFrame with 'EFFECT' and 'low_count'\n",
    "gene_df = pd.DataFrame({'GENE': gene_counts.index, 'freq_count': gene_counts.values})\n",
    "\n",
    "# Print the counts\n",
    "print(gene_df)\n",
    "\n",
    "# Save the DataFrame to a CSV file\n",
    "gene_df.to_csv('Low_CD4_results/impact_counts_low.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             GENE  freq_count\n",
      "0            rpoB          84\n",
      "1             rrl          70\n",
      "2            rpoC          64\n",
      "3           clpC1          57\n",
      "4            rpsA          49\n",
      "...           ...         ...\n",
      "1962         argC           1\n",
      "1963      Rv1639c           1\n",
      "1964        tesB1           1\n",
      "1965  Rv1610-trpC           1\n",
      "1966      Rv3922c           1\n",
      "\n",
      "[1967 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "# Count all unique values in the 'EFFECT' column\n",
    "gene_counts = low['Gene_id'].value_counts()\n",
    "\n",
    "# Create a DataFrame with 'EFFECT' and 'low_count'\n",
    "gene_df = pd.DataFrame({'GENE': gene_counts.index, 'freq_count': gene_counts.values})\n",
    "\n",
    "# Print the counts\n",
    "print(gene_df)\n",
    "gene_df['GENE'].value_counts().to_csv(\"Low_CD4_results/gene_counts_low.csv\")\n",
    "\n",
    "# Save the DataFrame to a CSV file\n",
    "#gene_df.to_csv('Low_CD4_results/gene_counts_low.csv', index=False)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
